### PR TITLE
add Taxonomic coverage

### PR DESCRIPTION
**Pull request by atn38**
see #25

### CODE REVIEW

**Comment by gastil** [comment]
I see no conflict with adding these tables as written here because the core mb does not already contain the taxonomic coverage tables that full mb has. We might consider also adding a self-referencing FK on the taxa list table so nesting can be used. That, if nesting can be offered without undue burden on those who wish to only list flat (un-nested) taxa. Also with EML2.2 we will have taxonomic authority ids for real, with uri. Will we be storing those s-a uri's in the semantic annotation schema or here with taxonomy? GCE had a separate schema for taxonomy because it can involve more complexity and have more purpose than just eml. If we might want to pare down the central schema (for bite sized implementation, or for high coherence), taxonomy and geography are good ones to spit out. But I digress...

**Comment by gastil** [comment]
We might consider having a cv_TaxonomicAuthority depending how much freedom for variation we want. What is to keep users from typing CAS one time and California Academy of Science the next time and the URI for that another time, not to mention miss-spelling and capitalization. Since it is part of the pk I feel it matters. I do not like to depend on the business logic layer to enforce constraints, esp when we  early adopters are direct-editing the sever tables (shh!).

**Comment by atn38** [comment]
Ha, that's embarrassing! I do remember running these, so something must
have change. Will resubmit.

**Comment by gastil** [comment]
No worries. I can edit and resubmit.

My plan is to make my own fork. Gastil-Buhl/lter_core_mb where I will
submit my pull requests for you to review.

Also I discovered why that one PK was missing in margaret's test install.
It is because only the first part, create tables, was run. then, since she
was not loading data into that one, the constraints script was not run.
None of the tables had their PK! So it was not any problem with the master
branch. Phew!

I'm dropping margaret's and making my own test install.

Now, go enjoy the SEV!

File: sql/08_create_taxonomic_table.sql [hunk 1]
```diff
@@ -0,0 +1,69 @@
+-- Note: FIRST: edit %db_owner% to the account name of the DB owner 
+
+CREATE TABLE lter_metabase."TaxaList" (
+	"TaxonID" character varying(50) NOT NULL;
+	"TaxonomicAuthority" character varying(50) NOT NULL;
+	"TaxonName" character varying(200);
```
Comment (gastil) [inline sql/08_create_taxonomic_table.sql]: the eml element names are taxonomicClassification taxonRankName taxonRankValue. And soon we will be able to also have the id. (@mob pls confirm we get both id and system, and these are going to be attributes of which element in eml2.2? Or I can look it up.) I want to use a consistent name mapping between column names of views and eml elements and attributes. I realize the names are long. But taxonname could be mistaken for a taxonomicRankValue without the Rank in there. A taxonRankName is one of {kingdom, phylum, class, order, family, genus, ...}, that is, the name of the rank. I want to keep rank in there not just for that clarity, but to stick to the pattern. I do not know if the R code follows that pattern. 
Aside from just good practice, it also avoids future name collisions.

File: sql/08_create_taxonomic_table.sql [hunk 2]
```diff
@@ -0,0 +1,69 @@
+-- Note: FIRST: edit %db_owner% to the account name of the DB owner 
+
+CREATE TABLE lter_metabase."TaxaList" (
+	"TaxonID" character varying(50) NOT NULL;
```
Comment (gastil) [inline sql/08_create_taxonomic_table.sql]: The extra semi-colons I'm editing out and will send so you (@atn38) can re-submit via git.

File: sql/08_create_taxonomic_table.sql (hunk 1 above)
Comment (atn38) [reply sql/08_create_taxonomic_table.sql]: There's been no R code written for taxonomy, so it's a blank canvas for now. Whenever we have the SQL view down, the R code will follow that.

**Comment by gastil** [comment]
(Not sure these notes should go in the pull request or an Issue.) 

Here is the EML2.2 snippet with the element and attribute names. I got it from this URL:

https://github.com/NCEAS/eml/blob/BRANCH_EML_2_2/docs/eml-220info.md

search-in-page for "Support for ids in <taxonomicClassification> element"

```
<taxonomicClassification id="taxon_MAPY">
   <taxonRankName>species</taxonRankName>
   <taxonRankValue>Macrocystis pyrifera</taxonRankValue>
   <commonName>Giant Kelp</commonName>
   <taxonId provider="ITIS">11274</taxonId>
   <taxonId provider="https://www.ncbi.nlm.nih.gov/taxonomy">35122</taxonId>
</taxonomicClassification>
```

where the external system is ITIS

The id in the taxonomicClassification attribute id="taxon_MAPY" is chosen by the user, in this case it was @mobb  for her favorite aquatic macrophyte, brown kelp. That id is not from ITIS of course. Looking at the EML2.1 schema I think I'm seeing that we already had that id. The new part is the repeatable element taxonId. It bothers me that the attribute provider can be an acronym or a uri. So, we'll have to accommodate both. (Why could those not have been separate attributes providerUri and providerCode? Blame me for not participating as much in EML2.2) Notice mb should offer a column for the commonName element. 

An we should re-write this together. A lot of changes.

**Comment by gastil** [comment]
The column LocalID in table TaxaList is for a site's own way of indexing taxa, such as the MAPY identifier for giant kelp at SBC. It is not necessary to fill in this column if your site does not use internal ids, since the id attribute of the element taxonomicClassification is optional in EML. Do not confuse TaxonID with the local ID. TaxonID is the element contents for the EML element taxonId. We opted to call the provider attribute taxon_id_provider in the R view rather than just provider (per An personal communication).

**Pull request merged**